### PR TITLE
Fix for #Pydev 1113: Create UI to select conda executable

### PR DESCRIPTION
**Pull request by oluiscabral**

### CODE REVIEW

File: plugins/org.python.pydev/src/org/python/pydev/ui/pythonpathconf/conda/PyDevCondaPreferences.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package org.python.pydev.ui.pythonpathconf.conda;
+
+import java.io.File;
+
+import org.python.pydev.core.IInterpreterInfo.UnableToFindExecutableException;
+import org.python.pydev.core.preferences.PydevPrefs;
+import org.python.pydev.ui.pythonpathconf.package_manager.CondaPackageManager;
+
+public final class PyDevCondaPreferences {
+    private static final String CONDA_EXEC_PATH = "CONDA_EXEC_PATH";
```
Comment (fabioz) [inline plugins/org.python.pydev/src/org/python/pydev/ui/pythonpathconf/conda/PyDevCondaPreferences.java:10]: I think this should be just '"CONDA_PATH" / "DEFAULT_CONDA_PATH".

File: plugins/org.python.pydev/src/org/python/pydev/ui/pythonpathconf/conda/PyDevCondaPreferences.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    private static final String DEFAULT_CONDA_EXEC_PATH = "";
+
+    public static final String getExecutablePath(CondaPackageManager condaPackageManager)
```
Comment (fabioz) [inline plugins/org.python.pydev/src/org/python/pydev/ui/pythonpathconf/conda/PyDevCondaPreferences.java:13]: I think that these setters would be a bit better receiving a File and returning a File and the getter should do a validation if the file exists (if the preference is set and the file doesn't exist it should throw a UnableToFindExecutableException).

**Review by fabioz — CHANGES_REQUESTED** [review]

File: plugins/org.python.pydev/src/org/python/pydev/ui/pythonpathconf/conda/PyDevCondaPreferences.java (hunk 2 above)
Comment (oluiscabral) [reply plugins/org.python.pydev/src/org/python/pydev/ui/pythonpathconf/conda/PyDevCondaPreferences.java:13]: So the setter should not have a file existance validation?

Comment (oluiscabral) [reply plugins/org.python.pydev/src/org/python/pydev/ui/pythonpathconf/conda/PyDevCondaPreferences.java:13]: And the setter should receive a file, set file path in cache and then return the same file?

Comment (fabioz) [reply plugins/org.python.pydev/src/org/python/pydev/ui/pythonpathconf/conda/PyDevCondaPreferences.java:13]: Yes, the setter doesn't need a validation (the getter needs to do it anyways -- for instance, it's possible that the user uninstalls conda and we have to handle that).

The getter should always read from the preference, convert the value to a `File`, validate and then return it.

Comment (oluiscabral) [reply plugins/org.python.pydev/src/org/python/pydev/ui/pythonpathconf/conda/PyDevCondaPreferences.java]: Update it in the last commit ;)

**Comment by fabioz** [comment]
This was merged in: https://github.com/fabioz/Pydev/commit/436abc536c5bbdf54b754f2488dd78f6a8c9b4bc

**Pull request closed without merging**